src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,343 @@
+            userInput = getUserInput(scan);
+        }
+
+        while (sayBye(userInput) == 0) {
+            if (userInput.contains(KEYWORD_LIST)) {
+                viewTasks(arrayOfTasks);
+            } else if (userInput.contains(KEYWORD_DONE)) {
+                markTaskAsDone(userInput);
+            } else if (userInput.contains(KEYWORD_DELETE)) {
+                deleteTask(userInput);
+            } else {
+                addTaskToArray(userInput);
+            }
+            // Update file
+            try {
+                updateFile(Path.of(fileName));
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+
+            // Get new userInput
+            userInput = getUserInput(scan);
+
+            // Clean text input, handle errors
+            while (invalidInput) {
+                userInput = getUserInput(scan);
+            }
+        }
+
+    }
+
+    private static void loadFile(String fileName) throws NoSuchFileException, IOException {
+        Scanner s = new Scanner(Path.of(fileName)); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String fileInput = s.nextLine();
+            String taskIcon = fileInput.substring(0, 3);
+            switch (taskIcon) {
+            case "[T]":
+                addTodoFromFile(fileInput);
+                break;
+            case "[E]":
+                addEventFromFile(fileInput);
+                break;
+            case "[D]":
+                addDeadlineFromFile(fileInput);
+                break;
+            default:
+                break;
+            }
+        }
+        viewTasks(arrayOfTasks);
+    }
+
+    private static void addDeadlineFromFile(String fileInput) {
+        int byIndex = fileInput.indexOf("(by:");
+        Deadline deadlineAdded = new Deadline(fileInput.substring(7, byIndex), fileInput.substring(byIndex + 5, fileInput.indexOf(")")));
+        arrayOfTasks.add(deadlineAdded);
+        if (fileInput.contains("\u2713")) {
+            arrayOfTasks.get(arrayOfTasks.size() - 1).markAsDone();
+        }
+    }
+
+    private static void addEventFromFile(String fileInput) {
+        int byIndex = fileInput.indexOf("(at:");
+        Event eventAdded = new Event(fileInput.substring(7, byIndex), fileInput.substring(byIndex + 5, fileInput.indexOf(")")));
+        arrayOfTasks.add(eventAdded);
+        if (fileInput.contains("\u2713")) {
+            arrayOfTasks.get(arrayOfTasks.size() - 1).markAsDone();
+        }
+    }
+
+    private static void addTodoFromFile(String fileInput) {
+        Todo todoAdded = new Todo(fileInput.substring(7));
+        arrayOfTasks.add(todoAdded);
+        if (fileInput.contains("\u2713")) {
+            arrayOfTasks.get(arrayOfTasks.size() - 1).markAsDone();
+        }
+    }
+
+    // Update file according to changes to list of tasks
+    private static void updateFile(Path filePath) throws IOException {
```

Comment:
Could consider creating a storage class and move these methods to the storage class.